pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,661 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/kcp-dev/logicalcluster"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	kuser "k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/handlers/negotiation"
+	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	coreexternalversions "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clusters"
+
+	"github.com/kcp-dev/kcp/pkg/apis/tenancy/projection"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	tenancyv1beta1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1beta1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpexternalversions "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+)
+
+const (
+	workspaceFullPathIndexName = "workspace-full-path"
+	homeOwnerRBACName          = "home-owner"
+	homeOwnerRBACPrefix        = "home-owner-"
+)
+
+var (
+	homeWorkspaceScheme = runtime.NewScheme()
+	homeWorkspaceCodecs = serializer.NewCodecFactory(homeWorkspaceScheme)
+)
+
+func init() {
+	_ = tenancyv1beta1.AddToScheme(homeWorkspaceScheme)
+}
+
+func WithHomeWorkspaces(apiHandler http.Handler, a authorizer.Authorizer, kubeClusterClient kubernetes.ClusterInterface, kcpClusterClient kcpclient.ClusterInterface, kubeSharedInformerFactory coreexternalversions.SharedInformerFactory, kcpSharedInformerFactory kcpexternalversions.SharedInformerFactory, creationDelay int, homePrefix logicalcluster.Name, bucketLevels, bucketSize int) http.Handler {
+	clusterWorkspaceInformer := kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces().Informer()
+	crInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoles().Informer()
+	crbInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings().Informer()
+
+	_ = addWorkspaceFullPathIndexer(clusterWorkspaceInformer)
+
+	return &homeWorkspaceHandler{
+		apiHandler:    apiHandler,
+		authz:         a,
+		creationDelay: creationDelay,
+		homePrefix:    homePrefix,
+		bucketLevels:  bucketLevels,
+		bucketSize:    bucketSize,
+		getLocalClusterWorkspace: func(fullName logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error) {
```

Comment:
what is a fullName?

Reply:
I removed all full name mentions and replaced with either `workspaceName` or `logicalClusterName` (or `lcluster`) in commit 1f37994d81146bb38fbecd97319d1a5a24eba461